### PR TITLE
bali-phy-2.3.7 (Update formula)

### PR DESCRIPTION
**Pull request by ge11232002**
Add the "desc" tag and "head"

### CODE REVIEW

File: bali-phy.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
 require 'formula'
 
 class BaliPhy < Formula
+  desc "Simultaneous Bayesian estimation of alignment and phylogeny"
   homepage 'http://www.bali-phy.org/'
-  url 'http://www.bali-phy.org/bali-phy-2.3.6.tar.gz'
-  sha1 '037db27894deb1df5fb7315376e7e1292c50f9c8'
+  url 'http://www.bali-phy.org/bali-phy-2.3.7.tar.gz'
+  sha256 '3120f9b448d308889093aa26a5941866c2841078e5eb9f1af1e2a6cbb134356a'
```
Comment (sjackman) [inline bali-phy.rb]: Can you also change the single quotes to double quotes? Try
`brew audit --strict bali-phy`
Thanks!

**Comment by tseemann** [comment]
```
Error Message

failed: brew audit Homebrew/science/bali-phy
Stacktrace
        Error: 1 problem in 1 formula
homebrew/science/bali-phy:
 * Non-executables were installed to "/usr/local/Cellar/bali-phy/2.3.7/bin"
The offending files are:
  /usr/local/Cellar/bali-phy/2.3.7/bin/compare-runs.R
  /usr/local/Cellar/bali-phy/2.3.7/bin/compare-runs2.R
  /usr/local/Cellar/bali-phy/2.3.7/bin/tree-plot1.R
  /usr/local/Cellar/bali-phy/2.3.7/bin/tree-plot2.R
  /usr/local/Cellar/bali-phy/2.3.7/bin/tree-plot3.R
```

**Comment by ge11232002** [comment]
These issues have been discussed in https://github.com/Homebrew/homebrew-science/pull/2019.
Do you have any better solution?

**Comment by tseemann** [comment]
Thank you for pointing me to #2019 

Usually these "auxiliary" or "helper" scripts are put into `libexec` and the master calling script is patched to point to `libexec` instead of `bin`. @iml ?

**Comment by tseemann** [comment]
I also notice it comes with BOOST bundled with it.
Should we try and use the Brew system BOOST?
```
--with-extra-libs=DIR   Add extra library paths
--with-system-boost     Use the Boost libraries installed on the system, instead of building our own.
```

**Comment by sjackman** [comment]
I prefer using the Brew dependency if possible. If the build systems make that difficult, I use the bundled dependency.

**Comment by ge11232002** [comment]
@sjackman Can you explain in more details? Thanks.

**Comment by sjackman** [comment]
I was saying that I prefer `--with-system-boost` over using a bundled version of boost.

**Comment by ge11232002** [comment]
Can I see the failed checks? The link to http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/2432/ is broken?

**Comment by tseemann** [comment]
@BrewTestBot test this please.

**Comment by sjackman** [comment]
http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/2435/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_Homebrew_science_bali_phy/
```
Testing homebrew/science/bali-phy
==> /usr/local/Cellar/bali-phy/2.3.7/bin/bali-phy /usr/local/Cellar/bali-phy/2.3.7/share/bali-phy/examples/Sequences/5S-rRNA/5d.fasta --iter=150
random seed = 16697502193367390750

bali-phy: Error! Failed to read from alignment-file '/usr/local/Cellar/bali-phy/2.3.7/share/bali-phy/examples/Sequences/5S-rRNA/5d.fasta --iter=150': file does not exist.
Error: homebrew/science/bali-phy: failed
Failed executing: /usr/local/Cellar/bali-phy/2.3.7/bin/bali-phy /usr/local/Cellar/bali-phy/2.3.7/share/bali-phy/examples/Sequences/5S-rRNA/5d.fasta\ --iter=150
```

**Comment by ge11232002** [comment]
It failed on mountain lion. I don't have one to test it... Any suggestions?

**Comment by sjackman** [comment]
Click `Details` beside `Some checks have failed`. It requires C++11.

http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/2497/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/install_Homebrew_science_bali_phy/
```
clang++ -DHAVE_CONFIG_H -I. -I../../src -I..  -I../../include -I../../src  -ffast-math -DNDEBUG -DNDEBUG_DP -funroll-loops  -Wall -Wextra -Wno-sign-compare -Woverloaded-virtual -Wstrict-aliasing -fvisibility=default -pipe -O3 -pedantic -I/usr/local/Cellar/libpng/1.6.17/include/libpng16 -I/usr/local/Cellar/freetype/2.6_1/include/freetype2 -I/usr/local/Cellar/fontconfig/2.11.1/include -I/usr/local/Cellar/freetype/2.5.3_1/include/freetype2 -I/usr/local/Cellar/pixman/0.32.6/include/pixman-1 -I/usr/local/Cellar/glib/2.44.1/include/glib-2.0 -I/usr/local/Cellar/glib/2.44.1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/cairo/1.14.2_1/include/cairo  -ftemplate-depth=256 -std=gnu++11 -c -o tree/sequencetree.o ../../src/tree/sequencetree.C
In file included from ../../src/likelihood.C:20:
In file included from ../../src/likelihood.H:23:
In file included from ../../src/imodel/imodel.H:29:
In file included from ../../src/models/model.H:36:
In file included from ../../src/bounds.H:30:
In file included from ../../src/util.H:651:
../../src/matrix.H:137:20: error: no member named 'move' in namespace 'std'
    operator=(std::move(m2));
              ~~~~~^
In file included from ../../src/rng.C:27:
../../src/rng.H:24:10: fatal error: 'random' file not found
#include <random>
         ^
1 error generated.
make[3]: *** [rng.o] Error 1
```

**Comment by sjackman** [comment]
The formula already has `needs :cxx11`. Try adding `ENV.cxx11`.

**Comment by ge11232002** [comment]
The build failed when using the system boost library on mountain lion with ENV.cxx11. It passed the check after using its own boost. I squashed the commits and the check failed. However, it doesn't seem to be my problem?

**Comment by sjackman** [comment]
Well that's an interesting error message.
http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/2507/version=yosemite/testReport/junit/brew-test-bot/yosemite/doctor/

```
Error Message

failed: brew doctor
Stacktrace

        Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:

    brew install zlib

Run `brew missing` for more details.
Warning: Homebrew's sbin was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting the PATH for example like so
    echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.bash_profile
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!
```

**Comment by ge11232002** [comment]
Yes.. Indeed. The build machine failed itself?

**Comment by sjackman** [comment]
Rebuilding…

**Comment by ge11232002** [comment]
Can it be merged now?

**Comment by sjackman** [comment]
Merged. Thanks for your patience, gtan.

**Pull request closed without merging**